airflow/utils/log/logging_mixin.py

```diff
@@ -73,11 +73,23 @@ def supports_external_link(self) -> bool:
         """Return whether handler is able to support external links."""
 
 
-class StreamLogWriter(IOBase):
+class StreamLogWriter(IO[str], IOBase):
     """Allows to redirect stdout and stderr to logger"""
 
     encoding: None = None
 
+    def fileno(self) -> int:
```

Comment:
Hmmm. I had the impression this has been here already :)

Reply:
I thought so too but it complained without it -- this was mostly around the use in StdoutRedirector which wants this to be `IO[str]`